UKTrains.indigoPlugin/Contents/Server Plugin/darwin_api.py

```
@@ -0,0 +1,185 @@
+	Retries up to 2 times with exponential backoff on API failures.
+
+	Args:
+		session: DarwinLdbSession object for API access
+		service_id: Unique service identifier
+
+	Returns:
+		ServiceDetails object or None if fetch failed after all retries
+	"""
+	try:
+		service = session.get_service_details(service_id)
+		return service
```

Ruff (0.14.14)

Consider moving this statement to an `else` block

(TRY300)